src/tls.c

```diff
@@ -120,10 +120,13 @@ static int ssl_seed(void)
  */
 int ssl_init()
 {
-  /* Load SSL and crypto error strings; register SSL algorithms */
-  SSL_load_error_strings();
+  /* OpenSSL library initialization
+   * If you are using 1.1.0 or above then you don't need to take any further steps. */
+#if OPENSSL_VERSION_NUMBER < 0x10100000L /* 1.1.0 */
   SSL_library_init();
-
+  SSL_load_error_strings();
+  OpenSSL_add_all_algorithms();
+#endif
```

Comment:
Shouldn't there be `#else OPENSSL_init_ssl()`? Where's the stuff that's called for 1.1.0 or higher?

Reply:
Thats the thing. For openbsd 1.1.0+ there is no initialization needed anymore :)
See also: https://wiki.openssl.org/index.php/Library_Initialization

Reply:
The manpage to OPENSSL_init_ssl also sais:

As of version 1.1.0 OpenSSL will automatically allocate all resources that it needs so no explicit
       initialisation is required. Similarly it will also automatically deinitialise as required